internal/daemon/worker/controller_connection_test.go

```diff
@@ -100,9 +98,6 @@ func TestMonitorUpstreamConnectionState(t *testing.T) {
 			case <-time.After(2 * time.Second):
 				t.Error("Time out waiting for condition")
 			}
-
-			got := upstreamConnectionState.Load()
-			assert.Equal(t, tt.expectedState, got)
```

Comment:
Is going from `READY` -> `CONNECTING` possible?

Reply:
Going from `READY` -> `CONNECTING` is not possible https://grpc.github.io/grpc/core/md_doc_connectivity-semantics-and-api.html

Going from `READY` -> `TRANSIENT_FAILURE` -> `CONNECTING` is possible but it can happen really quickly

Reply:
Interesting... then how is that flake failure occurring?